airbyte-cdk/bulk/core/base/src/main/kotlin/io/airbyte/cdk/command/FeatureFlag.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.command
+
+import io.micronaut.context.annotation.Factory
+import io.micronaut.context.env.Environment
+import jakarta.inject.Singleton
+import java.util.EnumSet
+
+/**
+ * An enum of all feature flags, currently these are set via environment vars.
+ *
+ * Micronaut can inject a Set<FeatureFlag> singleton of all active feature flags.
+ */
+enum class FeatureFlag(
+    val micronautEnvironmentName: String,
+    internal val envVar: EnvVar,
+    internal val requiredEnvVarValue: String,
```

Comment:
fyi I'm hoping to merge https://github.com/airbytehq/airbyte/pull/46363 soon, which adds a `docker run` version of DestinationProcess

specifically this line (https://github.com/airbytehq/airbyte/pull/46363/files#diff-3e529f9238c6c09b63750cc36fab22aef78c72fe55172ff6011244a4c41fd39fR108) wants the actual env var value.

Which I think works out fine, it just means this field should be public?

Reply:
It should work. Yes let's make all of this public.

Reply:
You can now replace your line with `FeatureFlag.AIRBYTE_CLOUD_DEPLOYMENT.envVarBindingDeclaration`